module/vision/GoalDetector/data/config/GoalDetector.yaml

```diff
@@ -1,8 +1,10 @@
-confidence_threshold: 0.7              # How confident do we need to be that this is a goal point
-cluster_points: 5                      # How many points do we need to claim this cluster is a viable goal
-disagreement_ratio: 0.20               # How much can the measured goal width vary from actual goal width before we
-                                       # treat the posts as part of separate goals
-goal_angular_cov: [0.1, 0.001, 0.001]  # Measurement certainties
-use_median: false                      # Use median rather than mean we determining position of left and right goal
-                                       # post edges
-debug: false                           # Turn on debug outputs
+confidence_threshold: 0.6             # How confident do we need to be that this is a goal point
+cluster_points: 5                     # How many points do we need to claim this cluster is a viable goal
+disagreement_ratio: 0.5               # How much can the measured goal width vary from actual goal width before we
+                                      # treat the posts as part of separate goals
+goal_angular_cov: [0.1, 0.001, 0.001] # Measurement certainties
+use_median: false                     # Use median rather than mean we determining position of left and right goal
+                                      # post edges
+max_goal_distance: 12                 # maximum distance that a goal may be from the robot to be a real goal,
```

Comment:
what's the rationale behind using the diagonal of the field?

Reply:
The largest distance that we could think of that should be allowed was diagonal, since you could be at the opposite end of the field (diagonal) to the goal post and you'd want to be able to see it. Any further and it'd be off the field.

Reply:
What is the logic behind those goal_angular_cov?

Reply:
Covariance is how certain we are of the measurements, basically magic numbers for the most part.